### PR TITLE
Activities unit tests and ai generation documentation

### LINKED ISSUE #173: ai gen documentation and unit testing
### Documentation Request
we need to be more transparent whenever ai is used to generate code. Furthermore, more unit testing is required.

### Purpose
increase tested code coverage and to be transparent about ai usage.

### Areas to Document
specific code snippets where ai was used.

### Audience
developers and professor grading project.

### Desired Format
inline comment

### Additional Context
ai comment will look like this:

//ai gen start (chat gpt 3.5, # representing human interference)
...
//ai gen stop

### PR DESCRIPTION
**Pull request by Ramirez-Christopher**
# Description

Adding unit testing to Activity side bar.

Fixes #173 

# Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/src/css/Activity.css
Comment (renm10) [inline frontend/src/css/Activity.css:1]: There actually exists a file called Activity.css.
It would be great if the file was renamed to Activities.css!

**Review by renm10 — CHANGES_REQUESTED** [review]
The unit tests for the activities look great, boosting the test coverage percentage!
AI documentations of your code is clear where it was used and how much.
The only problem persists in the Activity.css file where there exists a file with identical names. It would be distinguishable if you rename it to Activities.css. 
Otherwise, great work, everything else works amazing. I can approve the PR once you rename the file!

**Review by renm10 — APPROVED** [review]
The files are now distinguished with different names, and everything should be working fine!

**Review by axelauda — APPROVED** [review]
Just ran your unit test, good work on getting the coverage around 70-90% on those files.  This will help contribute to our overall coverage and hopefully we can get to at least 70%. Nice work!!

**Pull request merged**